src/runtime/resource/detail/detail_partitioner.cpp

```diff
@@ -916,7 +918,15 @@ namespace hpx { namespace resource { namespace detail
         cfg_.parse_result_ = cfg_.call(desc_cmdline, argc, argv);
 
         // set all parameters related to affinity data
-        pus_needed_ = affinity_data_.init(cfg_);
+        std::string affinity_description;
+        get_affinity_description(cfg_, affinity_description);
+
+        pus_needed_ = affinity_data_.init(cfg_.num_threads_,
+            hpx::util::safe_lexical_cast<std::size_t>(
+                get_config_entry("hpx.cores", 0), 0),
```

Comment:
Shouldn't this query the number of cores in a more general fashion (i.e. taking into account command line options, batch environment, etc.)?

Reply:
If I follow the logic correctly this does in fact include those things that you mention. The command line option (`--hpx:cores`) and batch environments are all handled together and the config entry `hpx.config` is updated to contain the value resulting from those. The behaviour is also unchanged from before.

Reply:
Ok, fair enough.